File: src/my_first_package/src/hello_loop.cpp
# include "rclcpp/rclcpp.hpp"

int main(int argc, char** argv){
    rclcpp::init(argc, argv);

    auto node = rclcpp::Node::make_shared("hello_loop");
    RCLCPP_INFO(node->get_logger(), "Hello, Before loop ROS2 world!");

    rclcpp::WallRate loop(1);
    while (rclcpp::ok())
    {
        RCLCPP_INFO(node->get_logger(), "Hello, ROS2 world!");
        loop.sleep();
    }
    
    rclcpp::shutdown();
    return 0;
}

File: python_ws/my_python_pkg/my_python_pkg/hello_loop.py
import rclpy
from time import sleep

def main(args=None):
    rclpy.init(args=args)
    node = rclpy.create_node("hello_loop")

    # keyboard 操作があるまでループ
    while rclpy.ok():
        node.get_logger().info("Hello, ROS2 world!")
        #rosのsleepがわからなかった。クラスならcallbackがあるっぽい
        sleep(1)

    node.destroy_node()
    rclpy.shutdown()

if __name__=="__main__":
    main()